### PR TITLE
Updated frontend-build to v12

### PR DESCRIPTION
**Pull request by BilalQamar95**
**Ticket:**
[42: Upgrade eslint to v8.x](https://github.com/openedx/frontend-wg/issues/42)

**What changed?**
- Updated `frontend-build `to v12 (Eslint was updated in `frontend-build` version resulting in it's version being bumped to v12. This PR updates `frontend-build` to reciprocate eslint version update)
- Resolved `eslint` issues post update

### CODE REVIEW

**Comment by arbrandes** [comment]
@BilalQamar95, do you mind resolving conflicts?  Thanks!

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -12,11 +12,13 @@ jobs:
     strategy:
       matrix:
         node: [16]
+        npm: [8.5.x]
```
Comment (leangseu-edx) [inline .github/workflows/validate.yml]: Do we need to check this? I think it is probably more reasonable to check just 8. Not all of node@16 correspond to npm@8.5.

**Comment by leangseu-edx** [comment]
There is recent change to upgrade `frontend-build@12`. https://github.com/openedx/frontend-app-learning/pull/1017/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519 I appreciate you taking your time to fix the linting. If you rebase this pr to master, I definitely can help getting this merge.

File: .github/workflows/validate.yml (hunk 1 above)
Comment (BilalQamar95) [reply .github/workflows/validate.yml]: This check was added to resolve `ci` dependency issues

Comment (leangseu-edx) [reply .github/workflows/validate.yml]: There seems to be some issue with your `package-lock.json` if the `ci` fail. For example, this pr no longer have `package.json` change and yet `package-lock.json` persist. I would suggest remove this and `package-lock.json`.

File: src/course-home/progress-tab/grades/grade-summary/GradeSummaryTable.jsx [hunk 2]
```diff
@@ -90,44 +100,31 @@ function GradeSummaryTable({ intl, setAllOfSomeAssignmentTypeIsLocked }) {
             Header: `${intl.formatMessage(messages.assignmentType)}`,
             accessor: 'type',
             // eslint-disable-next-line react/prop-types
```
Comment (leangseu-edx) [inline src/course-home/progress-tab/grades/grade-summary/GradeSummaryTable.jsx]:
```suggestion
```

File: src/course-home/progress-tab/grades/grade-summary/GradeSummaryTable.jsx [hunk 3]
```diff
@@ -90,44 +100,31 @@ function GradeSummaryTable({ intl, setAllOfSomeAssignmentTypeIsLocked }) {
...
-              />
-            ),
+            Cell: ({ value }) => getAssignmentTypeCell(value),
             headerClassName: 'h5 mb-0',
           },
           {
             Header: `${intl.formatMessage(messages.weight)}`,
             accessor: 'weight',
             headerClassName: 'justify-content-end h5 mb-0',
             // eslint-disable-next-line react/prop-types
```
Comment (leangseu-edx) [inline src/course-home/progress-tab/grades/grade-summary/GradeSummaryTable.jsx]:
```suggestion
```

File: src/course-home/progress-tab/grades/grade-summary/GradeSummaryTable.jsx [hunk 4]
```diff
@@ -90,44 +100,31 @@ function GradeSummaryTable({ intl, setAllOfSomeAssignmentTypeIsLocked }) {
...
-              <span className={value.locked ? 'greyed-out' : ''}>{value.weight}</span> // eslint-disable-line react/prop-types
-            ),
+            Cell: ({ value }) => getCell(value.locked, value.weight),
             cellClassName: 'text-right small',
           },
           {
             Header: `${intl.formatMessage(messages.grade)}`,
             accessor: 'grade',
             headerClassName: 'justify-content-end h5 mb-0',
             // eslint-disable-next-line react/prop-types
```
Comment (leangseu-edx) [inline src/course-home/progress-tab/grades/grade-summary/GradeSummaryTable.jsx]:
```suggestion
```

File: src/course-home/progress-tab/grades/grade-summary/GradeSummaryTable.jsx [hunk 5]
```diff
@@ -90,44 +100,31 @@ function GradeSummaryTable({ intl, setAllOfSomeAssignmentTypeIsLocked }) {
...
-              <span className={value.locked ? 'greyed-out' : ''}>{value.grade}</span> // eslint-disable-line react/prop-types
-            ),
+            Cell: ({ value }) => getCell(value.locked, value.grade),
             cellClassName: 'text-right small',
           },
           {
             Header: `${intl.formatMessage(messages.weightedGrade)}`,
             accessor: 'weightedGrade',
             headerClassName: 'justify-content-end h5 mb-0 text-right',
             // eslint-disable-next-line react/prop-types
```
Comment (leangseu-edx) [inline src/course-home/progress-tab/grades/grade-summary/GradeSummaryTable.jsx]:
```suggestion
```

**Review by leangseu-edx — APPROVED** [review]
I made a couple change by rebasing the change then remove `package-lock.json` and `.github/workflows/validate.yml`. I tested with everything working.

**Pull request merged**